packages/dev/docs/pages/react-aria/collections.mdx

```diff
@@ -0,0 +1,315 @@
+tables, trees, and grids. These collections can usually be navigated with the keyboard using arrow keys, and
+many have some form of selection. Many support loading data asynchronously, updating that data over time,
+virtualized scrolling for performance with large collections, and more.
+
+React Aria implements a JSX-based API for defining collections. This is an intuitive way to provide
+items with rich contents and various options as props. Building hierarchical
+collections, e.g. sections, or a tree of items is also very natural in JSX. React Aria provides a
+consistent API across many types of collection components that is easy to learn, performant with large
+collections, and extensible for advanced features.
+
+## Static collections
+
+A **static collection** is a collection that does not change over time (e.g. hard coded). This is common for
+components like action menus where the items are built into the application rather than representing user data.
+
+```tsx
+<Menu>
+  <MenuItem>Open</MenuItem>
+  <MenuItem>Edit</MenuItem>
+  <MenuItem>Delete</MenuItem>
+</Menu>
+```
+
+### Sections
+
+Sections or groups of items can be constructed by wrapping the items in a `<Section>` element. A `<Header>` can also be
+rendered within a `<Section>` to provide a section title.
+
+```tsx
+<Menu>
+  <Section>
+    <Header>Styles</Header>
+    <MenuItem>Bold</MenuItem>
+    <MenuItem>Underline</MenuItem>
+  </Section>
+  <Section>
+    <Header>Align</Header>
+    <MenuItem>Left</MenuItem>
+    <MenuItem>Middle</MenuItem>
+    <MenuItem>Right</MenuItem>
+  </Section>
+</Menu>
+```
+
+## Dynamic collections
+
+Static collections are great when the items never change, but how about dynamic data? A **dynamic collection**
+is a collection that is based on data, for example from an API. In addition, it may change over time as
+items are added, updated, or removed from the collection by a user.
+
+React Aria implements a JSX-based interface for dynamic collections, which maps over your data and
+applies a function for each item to render it. The following example shows how a collection can be
+rendered based on dynamic data, stored in React state.
+
+```tsx
+let [animals, setAnimals] = useState([
+  {id: 1, name: 'Aardvark'},
+  {id: 2, name: 'Kangaroo'},
+  {id: 3, name: 'Snake'}
+]);
+
+<ListBox items={animals}>
+  {item => <ListBoxItem>{item.name}</ListBoxItem>}
+</ListBox>
+```
+
+As you can see, the items are passed to the `items` prop of the top-level component, which iterates over each
+item and calls the function passed as children to the component. The item object is passed to the function, which
+returns a `<ListBoxItem>`.
+
+### Unique ids
+
+All items in a collection must have a unique id, which is used to determine what items in the collection changed
+when updates occur. By default, React Aria looks for an `id` or `key` property on each item object, which is often returned
```

Comment:
remove key?

Reply:
I think this is still correct. It's referring to the item objects, not the elements (e.g. `<ListBoxItem>`). We still look for both `id` and `key` as a property. https://github.com/adobe/react-spectrum/blob/9d8c74a344c9b7539f00fa58d0a02b062b6e830e/packages/react-aria-components/src/Collection.tsx#L665

Perhaps we can clarify this. Or do we want to keep supporting it?